servant-server/src/Servant/Server/Internal.hs

```diff
@@ -129,6 +129,13 @@ methodRouter :: (AllCTRender ctypes a)
 methodRouter method proxy status action = LeafRouter route'
   where
     route' request respond
+      | pathIsEmpty request && method == methodGet && requestMethod request == methodHead = do
+          runAction action respond $ \ output -> do
+            let accH = fromMaybe ct_wildcard $ lookup hAccept $ requestHeaders request
+            case handleAcceptH proxy (AcceptHeader accH) output of
+              Nothing -> failWith UnsupportedMediaType
+              Just (contentT, _) -> succeedWith $
+                responseLBS status [ ("Content-Type" , cs contentT)] ""
```

Comment:
Can we maybe abstract this code a little, given that it's almost exactly the same as the stuff just below?